File: src/main/java/com/better/alarm/model/AlarmCoreFactory.java
package com.better.alarm.model;

import com.better.alarm.configuration.Prefs;
import com.better.alarm.configuration.Store;
import com.better.alarm.logger.Logger;
import com.better.alarm.statemachine.HandlerFactory;

/**
 * Created by Yuriy on 09.08.2017.
 */

public class AlarmCoreFactory {
    private final Logger logger;
    private final IAlarmsScheduler alarmsScheduler;
    private final AlarmCore.IStateNotifier broadcaster;
    private final HandlerFactory handlerFactory;
    private final Prefs prefs;
    private final Store store;
    private final Calendars calendars;

    public AlarmCoreFactory(Logger logger, IAlarmsScheduler alarmsScheduler, AlarmCore.IStateNotifier broadcaster, HandlerFactory handlerFactory, Prefs prefs, Store store, Calendars calendars) {
        this.logger = logger;
        this.alarmsScheduler = alarmsScheduler;
        this.broadcaster = broadcaster;
        this.handlerFactory = handlerFactory;
        this.prefs = prefs;
        this.store = store;
        this.calendars = calendars;
    }

    public AlarmCore create(AlarmContainer container) {
        return new AlarmCore(container, logger, alarmsScheduler, broadcaster, handlerFactory, prefs, store, calendars);
    }
}

File: app/src/main/java/com/better/alarm/model/AlarmCoreFactory.kt
package com.better.alarm.model

import com.better.alarm.configuration.Prefs
import com.better.alarm.configuration.Store
import com.better.alarm.logger.Logger

/** Created by Yuriy on 09.08.2017. */
class AlarmCoreFactory(
    private val logger: Logger,
    private val alarmsScheduler: IAlarmsScheduler,
    private val broadcaster: AlarmCore.IStateNotifier,
    private val prefs: Prefs,
    private val store: Store,
    private val calendars: Calendars
) {
  fun create(container: AlarmStore): AlarmCore {
    return AlarmCore(container, logger, alarmsScheduler, broadcaster, prefs, store, calendars)
  }
}
